core/src/main/java/app/cash/paykit/core/impl/CashAppCashAppPayImpl.kt

```diff
@@ -321,15 +342,73 @@ internal class CashAppCashAppPayImpl(
         // If status is pending, schedule to check again.
         if (customerResponseData?.status == STATUS_PENDING) {
           // TODO: Add backoff strategy for long polling. ( https://www.notion.so/cashappcash/Implement-Long-pooling-retry-logic-a9af47e2db9242faa5d64df2596fd78e )
-          Thread.sleep(500)
+          try {
+            Thread.sleep(500)
+          } catch (e: InterruptedException) {
+            return@Thread
+          }
+
           poolTransactionStatus()
           return@Thread
         }
 
         // Unsuccessful transaction.
         setStateFinished(false)
       }
-    }.start()
+    }
+    try {
+      checkForApprovalThread?.start()
+    } catch (e: Exception) {
+      logError("Could not start checkForApprovalThread. Exception: $e")
+    }
+  }
+
+  private fun refreshUnauthorizedCustomerRequest(delaySeconds: Long) {
```

Comment:
It doesn't seem like we are checking for anything related to Unauthorized or expired here... but rather simply refreshing the customer request data repeatedly... 

```suggestion
  private fun refreshCustomerRequest(delaySeconds: Long) {
```